core/src/nats.rs

```diff
@@ -300,24 +296,46 @@ impl TypedNats {
 
         let consumer = stream
             .create_consumer(async_nats::jetstream::consumer::pull::Config {
-                deliver_policy: DeliverPolicy::Last,
+                deliver_policy,
                 filter_subject: subject.subject.clone(),
                 ..async_nats::jetstream::consumer::pull::Config::default()
             })
             .await
             .to_anyhow()?;
 
-        let result = match timeout(
-            Duration::from_secs(1),
-            consumer.stream().messages().await.to_anyhow()?.next(),
-        )
-        .await
-        {
-            Ok(Some(v)) => v.to_anyhow()?,
-            _ => return Ok(None),
-        };
-
-        Ok(Some(serde_json::from_slice(&result.payload)?))
+        let mut result: Vec<T> = Vec::new();
+
+        loop {
+            let mut messages = consumer.fetch().messages().await.to_anyhow()?;
+            let mut done = true;
+
+            while let Some(v) = messages.next().await {
+                let v = v.to_anyhow()?;
+                done = false;
+
+                result.push(serde_json::from_slice(&v.payload)?);
+
+                if v.info().to_anyhow()?.pending == 0 {
+                    done = true;
+                    break
+                }                
+            }
+
+            if done {
+                break;
```

Comment:
If a subject is having messages continually delivered to it, and I call get_all, will it ever terminate?

Reply:
Good point, I think it would not terminate.

I guess we could look at the first "pending" count, and use that. I'll rewrite again.

Reply:
That solves the error issue too no?

Reply:
Not entirely, because pending is an attribute of messages themselves. If there are no messages in the stream the error occurs before we have a chance to read a `.pending` value.

I've decided that this is something I'd like to look at once I have my head around streams a bit more, so I've replaced it with a TODO. For our use case the messages are low-frequency enough for this not to happen.